include/f_core/os/c_datalogger.h

```diff
@@ -0,0 +1,75 @@
+#ifndef C_DATALOGGER_H
+#define C_DATALOGGER_H
+
+#include <cstdint>
+#include <type_traits>
+#include <zephyr/fs/fs.h>
+
+enum class LogMode { Growing, Circular, FixedSize };
+
+/// @brief Internal, type-unsafe datalogger. Don't use this directly. Use CDataLogger instead
+namespace detail {
+class datalogger {
```

Comment:
Can we namespace it NDetail and name the class CPrivateDataLogger?

Reply:
suppose it could be. a namespace called `detail` is a common approach to say don't use this directly which is why I didn't give it the N. But its all agreed upon standards so we can just as easily agree upon our own.